Hamburger menu UI mobile

renahashimi opened the pull request:
**Pull Request Summary for #156** 
- **Toggle Functionality for Menus:** Updated toggleHamburgerMenu to manage the visibility of the hamburger navigation and file menus.
- **Button Hover Effects:** Enhanced button interactions with smooth transitions and increased font size for active buttons.
- **Mobile Navigation Improvements:** Updated styles for a more streamlined and user-friendly design.
- **Updated index.html:** Fixed errors and ensured proper closing tags (note: further cleanup may be needed).
- **Update vite.config.js:** Although this was not part of the task, I edited it to avoid errors during commit by removing unnecessary error text from the configuration file to improve clarity and maintainability.

Commit:
Added toggle functionality for file menu and hamburger menu

- Implemented the `toggleHamburgerMenu` function to handle toggling the hamburger navigation and file menu visibility.
- The hamburger button now toggles the visibility of the navigation menu.
- The file button toggles the visibility of the file menu and adds/removes an active state.
- Added logic to close the menus when clicking outside them.

Commit:
Improve button hover effects and active state

- Added smooth transitions and hover scaling for buttons.
- Increased font size for active buttons with `.active` class.

Commit:
Improve mobile navigation appearance.

- Updated styles for a more streamlined and user-friendly design.
- To commit this, I had to remove som errors  and add closing </div> - // prettier

Commit:
Update vite.config.js

- Error text in file - from workflow - removed and commited.

Commit:
Update index.html

- Added the background color to zoom
- Updated minor elements to prevent any disruption to the md-size.

jonhenrikaavitsland reviewed (changes requested):
@renahashimi I need you to resolve the merge conflicts. Please request a new review when it has been done.

Commit:
Resolving conflicts

Commit:
Update index.css

Commit:
Update index.css

Commit:
Resolve merge conflict - index.html

Commit:
Update style in NAV

Commit:
Update HTML IDs and simplify JavaScript for hamburger menu toggle

- Changed ID of the mobile navigation div to "mobileNav" // new workflow index.html.
- Updated JavaScript to streamline event handling for the hamburger menu and file menu.

Commit:
Update index.html

jonhenrikaavitsland reviewed (changes requested):
@RamonaXR Can you please fix the conflicts and request my review after

jonhenrikaavitsland commented:
@sulenchy The Dev has issues trying to fix the merge conflicts, I was trying to assist but we was never able to solve the conflicts properly. If it is possible with some assistance on this, we would appreciate it.

Commit:
Merge remote-tracking branch 'origin/workflow' into hamburger-menu-ui-mobile-update

Commit:
Resolving merge conflict #181

Had to update package to remove errors.

Commit:
Update package-lock.json

jonhenrikaavitsland reviewed (approved):
@renahashimi thank you. 
The conflicts has been resolved.

jonhenrikaavitsland commented:
@renahashimi merged successful, good job.
